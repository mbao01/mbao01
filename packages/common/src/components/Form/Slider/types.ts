import * as SliderPrimitive from "@radix-ui/react-slider";
import { type VariantProps } from "../../../libs";
import {
  getSliderClasses,
  getSliderRootClasses,
  getSliderThumbClasses,
  getSliderTrackClasses,
} from "./constants";

export type SliderProps = React.ComponentPropsWithoutRef<typeof SliderPrimitive.Root> &
  VariantProps<typeof getSliderClasses> &
  VariantProps<typeof getSliderTrackClasses> &
  VariantProps<typeof getSliderThumbClasses> &
  VariantProps<typeof getSliderRootClasses>;
