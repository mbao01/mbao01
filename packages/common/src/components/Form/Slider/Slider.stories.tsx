import type { Meta, StoryObj } from "@storybook/react";
import { FormControl } from "../components/FormControl";
import { Label } from "../components/Label";
import { Slider } from "./Slider";

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
  component: Slider,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
    layout: "centered",
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {},
} satisfies Meta<typeof Slider>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
  args: {
    defaultValue: [50],
    max: 100,
    step: 1,
  },
};

export const PrimarySlider: Story = {
  args: {
    variant: "primary",
  },
};

export const TinySlider: Story = {
  args: {
    size: "xs",
  },
};

export const DisabledSlider: Story = {
  args: {
    disabled: true,
    defaultValue: [25],
  },
};

export const LabeledSlider: Story = {
  render: (args) => {
    return (
      <FormControl as="div" className="flex-row items-center">
        <Slider id="terms" {...args} />
        <Label htmlFor="terms">Risk Level</Label>
      </FormControl>
    );
  },
};
