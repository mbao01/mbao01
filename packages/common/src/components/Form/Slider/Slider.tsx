"use client";

import * as React from "react";
import * as SliderPrimitive from "@radix-ui/react-slider";
import { cn } from "../../../utilities";
import {
  getSliderClasses,
  getSliderRootClasses,
  getSliderThumbClasses,
  getSliderTrackClasses,
} from "./constants";
import { type SliderProps } from "./types";

const Slider = React.forwardRef<
  React.ElementRef<typeof SliderPrimitive.Root>,
  SliderProps
>(({ className, size, variant, wide, disabled, name, ...props }, ref) => (
  <SliderPrimitive.Root
    ref={ref}
    name={name}
    disabled={disabled}
    className={cn(getSliderRootClasses({ wide, disabled }), className)}
    data-testid={`${name}-slider-root`}
    {...props}
  >
    <SliderPrimitive.Track
      className={cn(getSliderTrackClasses({ size, variant }))}
      data-testid={`${name}-slider-track`}
    >
      <SliderPrimitive.Range
        className={cn(getSliderClasses({ variant }))}
        data-testid={`${name}-slider-range`}
      />
    </SliderPrimitive.Track>
    <SliderPrimitive.Thumb
      className={cn(getSliderThumbClasses({ size, variant }))}
      data-testid={`${name}-slider-thumb`}
    />
  </SliderPrimitive.Root>
));
Slider.displayName = SliderPrimitive.Root.displayName;

export { Slider };
