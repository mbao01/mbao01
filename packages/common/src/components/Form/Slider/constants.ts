import { cva } from "../../../libs";

export const getSliderRootClasses = cva(
  "relative flex touch-none select-none items-center min-w-24",
  {
    variants: {
      disabled: {
        true: "cursor-not-allowed opacity-80",
      },
      wide: {
        true: "w-full",
      },
    },
  }
);

export const getSliderTrackClasses = cva("relative h-2 w-full grow overflow-hidden rounded-full", {
  variants: {
    variant: {
      primary: "bg-neutral-content",
      secondary: "bg-neutral-content",
      accent: "bg-neutral-content",
      success: "bg-neutral-content",
      warning: "bg-neutral-content",
      info: "bg-neutral-content",
      error: "bg-neutral-content",
      neutral: "bg-base-200",
    },
    size: {
      xs: "h-0.5",
      sm: "h-1",
      md: "h-2",
      lg: "h-3",
    },
  },
  defaultVariants: {
    variant: "neutral",
  },
});

export const getSliderClasses = cva("absolute h-full", {
  variants: {
    variant: {
      primary: "bg-primary",
      secondary: "bg-secondary",
      accent: "bg-accent",
      success: "bg-success",
      warning: "bg-warning",
      info: "bg-info",
      error: "bg-error",
      neutral: "bg-base-content",
    },
  },
  defaultVariants: {
    variant: "neutral",
  },
});

export const getSliderThumbClasses = cva(
  "block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50",
  {
    variants: {
      variant: {
        primary: "border-primary bg-base-100",
        secondary: "border-secondary bg-base-100",
        accent: "border-accent bg-base-100",
        success: "border-success bg-base-100",
        warning: "border-warning bg-base-100",
        info: "border-info bg-base-100",
        error: "border-error bg-base-100",
        neutral: "border-neutral-content bg-neutral",
      },
      size: {
        xs: "h-4 w-4",
        sm: "h-4 w-4",
        md: "h-5 w-5",
        lg: "h-7 w-7",
      },
    },
    defaultVariants: {
      variant: "neutral",
    },
  }
);
