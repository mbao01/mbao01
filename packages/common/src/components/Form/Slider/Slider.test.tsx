import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { Slider } from "./Slider";

describe("Slider", () => {
  it("renders with the correct initial state", () => {
    const { asFragment } = render(<Slider name="risk-level" />);

    expect(screen.getByRole("slider")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([
    "primary",
    "secondary",
    "accent",
    "success",
    "warning",
    "info",
    "error",
    "neutral",
  ] as const)("has %s variant", (variant) => {
    const { asFragment } = render(
      <Slider name="risk-level" variant={variant} wide />
    );

    expect(screen.getByRole("slider")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it.each([
    { size: "xs", name: "tiny" },
    { size: "sm", name: "small" },
    { size: "md", name: "base" },
    { size: "lg", name: "large" },
  ] as const)("has $name ($size)", ({ size, name }) => {
    const { asFragment } = render(<Slider name={name} size={size} />);

    expect(screen.getByRole("slider")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
