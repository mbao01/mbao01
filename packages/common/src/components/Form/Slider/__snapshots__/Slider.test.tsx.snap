// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slider > has 'base' ('md') 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24"
    data-orientation="horizontal"
    data-testid="base-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative w-full grow overflow-hidden rounded-full bg-base-200 h-2"
      data-orientation="horizontal"
      data-testid="base-slider-track"
    >
      <span
        class="absolute h-full bg-base-content"
        data-orientation="horizontal"
        data-testid="base-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-neutral-content bg-neutral h-5 w-5"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="base-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has 'large' ('lg') 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24"
    data-orientation="horizontal"
    data-testid="large-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative w-full grow overflow-hidden rounded-full bg-base-200 h-3"
      data-orientation="horizontal"
      data-testid="large-slider-track"
    >
      <span
        class="absolute h-full bg-base-content"
        data-orientation="horizontal"
        data-testid="large-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-neutral-content bg-neutral h-7 w-7"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="large-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has 'small' ('sm') 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24"
    data-orientation="horizontal"
    data-testid="small-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative w-full grow overflow-hidden rounded-full bg-base-200 h-1"
      data-orientation="horizontal"
      data-testid="small-slider-track"
    >
      <span
        class="absolute h-full bg-base-content"
        data-orientation="horizontal"
        data-testid="small-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-neutral-content bg-neutral h-4 w-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="small-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has 'tiny' ('xs') 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24"
    data-orientation="horizontal"
    data-testid="tiny-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative w-full grow overflow-hidden rounded-full bg-base-200 h-0.5"
      data-orientation="horizontal"
      data-testid="tiny-slider-track"
    >
      <span
        class="absolute h-full bg-base-content"
        data-orientation="horizontal"
        data-testid="tiny-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-neutral-content bg-neutral h-4 w-4"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="tiny-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has accent variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-neutral-content"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-accent"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-accent bg-base-100"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has error variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-neutral-content"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-error"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-error bg-base-100"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has info variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-neutral-content"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-info"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-info bg-base-100"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has neutral variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-base-200"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-base-content"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-neutral-content bg-neutral"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has primary variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-neutral-content"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-primary"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-primary bg-base-100"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has secondary variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-neutral-content"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-secondary"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-secondary bg-base-100"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has success variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-neutral-content"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-success"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-success bg-base-100"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > has warning variant 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24 w-full"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-neutral-content"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-warning"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-warning bg-base-100"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;

exports[`Slider > renders with the correct initial state 1`] = `
<DocumentFragment>
  <span
    aria-disabled="false"
    class="relative flex touch-none select-none items-center min-w-24"
    data-orientation="horizontal"
    data-testid="risk-level-slider-root"
    dir="ltr"
    style="--radix-slider-thumb-transform: translateX(-50%);"
  >
    <span
      class="relative h-2 w-full grow overflow-hidden rounded-full bg-base-200"
      data-orientation="horizontal"
      data-testid="risk-level-slider-track"
    >
      <span
        class="absolute h-full bg-base-content"
        data-orientation="horizontal"
        data-testid="risk-level-slider-range"
        style="left: 0%; right: 100%;"
      />
    </span>
    <span
      style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
    >
      <span
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="block h-5 w-5 rounded-full border-2 ring-offset-neutral transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border-neutral-content bg-neutral"
        data-orientation="horizontal"
        data-radix-collection-item=""
        data-testid="risk-level-slider-thumb"
        role="slider"
        style=""
        tabindex="0"
      />
    </span>
  </span>
</DocumentFragment>
`;
